src/cpu/rv64/rvv_inner_product.hpp

```diff
@@ -0,0 +1,155 @@
+                    VERBOSE_UNSUPPORTED_TAG);
+            VDISPATCH_INNER_PRODUCT(
+                    attr_.set_default_formats(dst_md(0)) == status::success,
+                    VERBOSE_UNSUPPORTED_TAG);
+
+            const memory_desc_wrapper src_d(src_md(0));
+            const memory_desc_wrapper wei_d(weights_md(0));
+            const memory_desc_wrapper dst_d(dst_md(0));
+
+            VDISPATCH_INNER_PRODUCT(check_layouts(src_d, wei_d, dst_d),
+                    VERBOSE_UNSUPPORTED_TAG);
+
+            // Book scratchpad for contiguous packing of src/weights per-thread
+            {
+                auto scratchpad = scratchpad_registry().registrar();
+                const dim_t K
+                        = (dim_t)IC() * (dim_t)KD() * (dim_t)KH() * (dim_t)KW();
+                const size_t src_elt = types::data_type_size(src_type);
+                const size_t wei_elt = types::data_type_size(wei_type);
+                const int nthr = dnnl_get_max_threads();
+                scratchpad.book(memory_tracking::names::key_iprod_src_reorder,
+                        (size_t)K * (size_t)nthr, src_elt);
+                scratchpad.book(
+                        memory_tracking::names::key_iprod_weights_reorder,
+                        (size_t)K * (size_t)nthr, wei_elt);
+            }
+
+            return status::success;
+        }
+
+        bool check_types(const data_type_t &src_type,
+                const data_type_t &wei_type, const data_type_t &dst_type,
+                const data_type_t &bia_type) const {
+            using namespace data_type;
+            const bool dst_ok = utils::one_of(dst_type, f32, s32, s8, u8);
+            const bool src_wei_ok = (src_type == f32 && wei_type == f32)
+                    || (src_type == s8 && wei_type == s8)
+                    || (src_type == u8 && wei_type == s8);
+            const bool bia_ok = IMPLICATION(
+                    with_bias(), utils::one_of(bia_type, f32, src_type));
+            return dst_ok && src_wei_ok && bia_ok
+                    && platform::has_data_type_support(src_type)
+                    && platform::has_data_type_support(wei_type)
+                    && platform::has_data_type_support(dst_type);
+        }
+
+        bool check_layouts(const memory_desc_wrapper &src_d,
+                const memory_desc_wrapper &wei_d,
+                const memory_desc_wrapper &dst_d) const {
+            // Supported memory layouts: plain, dense, no inner blocks
+            const bool plain_dense = src_d.blocking_desc().inner_nblks == 0
+                    && wei_d.blocking_desc().inner_nblks == 0
+                    && dst_d.blocking_desc().inner_nblks == 0
+                    && src_d.is_dense(/*with_padding=*/false)
+                    && wei_d.is_dense(/*with_padding=*/false)
+                    && dst_d.is_dense(/*with_padding=*/false)
+                    && src_d.is_plain() && wei_d.is_plain() && dst_d.is_plain();
```

Comment:
is_plain is equal to inner_nblks == 0, see [here].(https://github.com/uxlfoundation/oneDNN/blob/main/src/common/memory_desc_wrapper.hpp#L462-L465)